hack/update-ubuntu-image.sh

```diff
@@ -0,0 +1,324 @@
+#!/usr/bin/env bash
+
+function print_help() {
+	cat <<HELP
+$(basename "${BASH_SOURCE[0]}"): Update the Ubuntu image location in the specified templates
+
+Usage:
+  $(basename "${BASH_SOURCE[0]}") [--minimal|--server] [--version <version>] <template.yaml>...
+
+Description:
+  This script updates the Ubuntu image location in the specified templates.
+  If the image location in the template contains a release date in the URL, the script replaces it with the latest available date.
+  If no flags are specified, the script uses the flavor and version from the image location basename in the template.
+
+  Image location basename format: ubuntu-<version>-<flavor>-cloudimg-<arch>.img
+
+  Released Ubuntu image information is fetched from the following URLs:
+
+    Server: https://cloud-images.ubuntu.com/releases/stream/v1/com.ubuntu.cloud:released:download.json
+    Minimal: https://cloud-images.ubuntu.com/minimal/releases/stream/v1/com.ubuntu.cloud:released:download.json
+
+  The downloaded JSON file will be cached in the Lima cache directory.
+
+Examples:
+  Update the Ubuntu image location in templates/**.yaml:
+  $ $(basename "${BASH_SOURCE[0]}") templates/**.yaml
+
+  Update the Ubuntu image location in ~/.lima/ubuntu/lima.yaml:
+  $ $(basename "${BASH_SOURCE[0]}") ~/.lima/ubuntu/lima.yaml
+
+  Update the Ubuntu image location to ubuntu-24.04-minimal-cloudimg-<arch>.img in ~/.lima/docker/lima.yaml:
+  $ $(basename "${BASH_SOURCE[0]}") --minimal --version 24.04 ~/.lima/docker/lima.yaml
+
+Flags:
+  --server             Use the server flavor image
+  --minimal            Use the minimal flavor image
```

Comment:
This should be like `--flavor=(server|minimal)`)?
More "flavors" can be potentially added, especially when adding the support for non-Ubuntu distros.

Does `minimal` flavor really work?

Reply:
> Does `minimal` flavor really work?

As far as I’ve tested, `docker.yaml` works fine even when switched to minimal. Any issues have already been fixed in the past (#2440).

Reply:
> This should be like --flavor=(server|minimal))?

added `--flavor <flavor>` flag.